src/Illuminate/Filesystem/FilesystemAdapter.php

```diff
@@ -64,13 +64,20 @@ public function get($path)
 	 * Write the contents of a file.
 	 *
 	 * @param  string  $path
-	 * @param  string  $contents
+	 * @param  string|resource  $contents
 	 * @param  string  $visibility
 	 * @return bool
 	 */
 	public function put($path, $contents, $visibility = null)
 	{
-		return $this->driver->put($path, $contents, ['visibility' => $this->parseVisibility($visibility)]);
+		if (is_resource($contents))
+		{
+			return $this->driver->putStream($path, $contents, ['visibility' => $this->parseVisibility($visibility)]);
```

Comment:
@jhdxr I believe he is referring to this, `['visibility' => $this->parseVisibility($visibility)]`. I would also remove the `else { }` surrounding the final return statement.